core/src/main/java/org/apache/iceberg/hadoop/HadoopMetricsContext.java

```diff
@@ -49,7 +49,7 @@ public HadoopMetricsContext(String scheme) {
   public void initialize(Map<String, String> properties) {
     // FileIO has no specific implementation class, but Hadoop will
     // still track and report for the provided scheme.
-    this.scheme = properties.getOrDefault(SCHEME, scheme);
+    this.scheme = properties.getOrDefault(CatalogProperties.IO_METRICS_SCHEME, scheme);
```

Comment:
I think we no longer need constructor `public HadoopMetricsContext(String scheme)`, we should use a no-arg constructor because we are already initializing the scheme here. The scheme should be passed in from S3FileIO.